src/spec/OpenApi.php

```diff
@@ -79,4 +79,47 @@ public function performValidation()
             $this->addError('Unsupported openapi version: ' . $this->openapi);
         }
     }
+
+    /**
+     * Thanks https://www.php.net/manual/en/function.array-merge-recursive.php#96201
+     *
+     * Merges any number of arrays / parameters recursively, replacing
+     * entries with string keys with values from latter arrays.
+     * If the entry or the next value to be assigned is an array, then it
+     * automagically treats both arguments as an array.
+     * Numeric entries are appended, not replaced, but only if they are
+     * unique
+     *
+     * Function call example: `$result = array_merge_recursive_distinct(a1, a2, ... aN);`
+     * More documentation is present at [array merge recursive distinct.md](../../../doc/array merge recursive distinct.md) file
+     */
+    public static function arrayMergeRecursiveDistinct()
```

Comment:
this should be a method of a separate helper class.

Reply:
Implemented in 81ee4edb6882943f846d95f37f3d26e42eb66662